src/controllers/FacebookController.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Facebook-specific endpoints live here
+ */
+
+use GuzzleHttp\Client;
+
+class FacebookController extends ApiController
+{
+    /**
+     * Take the verification code from the client, send to Facebook to get an access token.
+     * With the access token, read the user's profile to get their email address.
+     * From that, look up who they are, create them a new access token and return
+     * the same format as we do when logging in a user
+     */
+    public function logUserIn($request, $db)
+    {
+        $clientId = $request->getParameter('client_id');
+        $clientSecret = $request->getParameter('client_secret');
+        $this->oauthModel = $request->getOauthModel($db);
+        if (!$this->oauthModel->isClientPermittedPasswordGrant($clientId, $clientSecret)) {
+            throw new Exception("This client cannot perform this action", 403);
+        }
+
+        // check incoming values
+        if (empty($code = $request->getParameter("code"))) {
+            throw new Exception("The request code must be supplied");
+        }
+
+        // exchange code for access token
+        $client = new Client([
+            'headers' => ['Accept' => 'application/json']
+        ]);
+
+        $res = $client->get('https://graph.facebook.com/v2.3/oauth/access_token', [
```

Comment:
For just hitting the a couple of endpoints, pulling in the entire Facebook SDK seems a little like overkill to me.